src/main/java/org/opensearch/ad/transport/GetAnomalyDetectorTransportAction.java

```diff
@@ -1,430 +1,438 @@
+    private final Set<DetectorProfileName> allProfileTypes;
+    private final Set<DetectorProfileName> defaultDetectorProfileTypes;
+    private final Set<String> allEntityProfileTypeStrs;
+    private final Set<EntityProfileName> allEntityProfileTypes;
+    private final Set<EntityProfileName> defaultEntityProfileTypes;
+    private final NamedXContentRegistry xContentRegistry;
+    private final DiscoveryNodeFilterer nodeFilter;
+    private final TransportService transportService;
+    private volatile Boolean filterByEnabled;
+    private final ADTaskManager adTaskManager;
+
+    @Inject
+    public GetAnomalyDetectorTransportAction(
+        TransportService transportService,
+        DiscoveryNodeFilterer nodeFilter,
+        ActionFilters actionFilters,
+        SDKClusterService clusterService,
+        SDKRestClient client,
+        Settings settings,
+        NamedXContentRegistry xContentRegistry,
+        ADTaskManager adTaskManager
+    ) {
+        // super(GetAnomalyDetectorAction.NAME, transportService, actionFilters, GetAnomalyDetectorRequest::new);
+        this.clusterService = clusterService;
+        this.client = client;
+
+        List<DetectorProfileName> allProfiles = Arrays.asList(DetectorProfileName.values());
+        this.allProfileTypes = EnumSet.copyOf(allProfiles);
+        this.allProfileTypeStrs = getProfileListStrs(allProfiles);
+        List<DetectorProfileName> defaultProfiles = Arrays.asList(DetectorProfileName.ERROR, DetectorProfileName.STATE);
+        this.defaultDetectorProfileTypes = new HashSet<DetectorProfileName>(defaultProfiles);
+
+        List<EntityProfileName> allEntityProfiles = Arrays.asList(EntityProfileName.values());
+        this.allEntityProfileTypes = EnumSet.copyOf(allEntityProfiles);
+        this.allEntityProfileTypeStrs = getProfileListStrs(allEntityProfiles);
+        List<EntityProfileName> defaultEntityProfiles = Arrays.asList(EntityProfileName.STATE);
+        this.defaultEntityProfileTypes = new HashSet<EntityProfileName>(defaultEntityProfiles);
+
+        this.xContentRegistry = xContentRegistry;
+        this.nodeFilter = nodeFilter;
+        filterByEnabled = AnomalyDetectorSettings.FILTER_BY_BACKEND_ROLES.get(settings);
+        try {
+            clusterService.getClusterSettings().addSettingsUpdateConsumer(FILTER_BY_BACKEND_ROLES, it -> filterByEnabled = it);
+        } catch (Exception e) {
+            // FIXME Handle this
+            // https://github.com/opensearch-project/opensearch-sdk-java/issues/422
+        }
+
+        this.transportService = transportService;
+        this.adTaskManager = adTaskManager;
+    }
+
+    public void doExecute(Task task, GetAnomalyDetectorRequest request, ActionListener<GetAnomalyDetectorResponse> actionListener) {
+        String detectorID = request.getDetectorID();
+        // Temporary null user for AD extension without security. Will always execute detector.
+        UserIdentity user = getNullUser();
```

Comment:
interesting, Ill make sure to bypass user resolution this way for `start detector`